doc/api/test.md

```diff
@@ -3390,6 +3394,16 @@ changes:
 -->
 
 * `count` {number} The number of assertions and subtests that are expected to run.
+* `options` {Object} Additional options for the plan.
+  * `wait` {boolean|number} The wait time for the plan:
+    * If `true`, the plan waits indefinitely for all assertions and subtests to run.
+    * If `false`, the plan performs an immediate check after the test function completes,
+      without waiting for any pending assertions or subtests.
+      Any assertions or subtests that complete after this check will not be counted towards the plan.
+    * If a number, it specifies the maximum wait time in milliseconds
+      before timing out while waiting for expected assertions and subtests to be matched.
+      If the timeout is reached, the test will fail.
```

Comment:
Should the maximal value be pointed out here?

Reply:
Are you suggesting that there should be a maximum value possible for the wait?

Reply:
@jsumners, yes, at the moment the maximum is `const { TIMEOUT_MAX } = require('internal/timers');`, here a partial reference https://github.com/nodejs/node/issues/11198

Reply:
I don't see a need for it. If someone wants to set the timeout to `Number.MAX_VALUE`, then that's on them. I think most people will be reasonable and set it on the order of seconds.

Reply:
IMHO, validating the maximum value is a good way to inform the user of misuse.
I honestly would avoid reporting this maximum in the documentation, as has been done in this case https://github.com/nodejs/node/pull/56595/files .

Reply:
@atlowChemi @jsumners If you are okay with this, I would avoid documenting this implementation detail!

Reply:
I'll just address the last comment related to tests and restart the CIs

Reply:
@pmarchini I left this comment while approving the PR - so this is non-blocking IMHO 🙂